ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/transactions/PendingTransactions.java

```diff
@@ -51,10 +56,11 @@
  */
 public class PendingTransactions {
 
-  public static final int MAX_PENDING_TRANSACTIONS = 4096;
-  public static final int DEFAULT_TX_RETENTION_HOURS = 13;
+  private static final Logger LOG = LogManager.getLogger();
+  public static final int MAX_PENDING_TRANSACTIONS = 1024;
+  public static final int DEFAULT_TX_RETENTION_SECONDS = 60;
```

Comment:
I think this change will be problematic for local transaction propagation.  It can take us a while to find peers, so a 1 minute time window could mean local transactions get dropped before we have a chance to broadcast them to anyone.

Reply:
You are right. Should we have different eviction policies for local and remote transactions ?

Reply:
If there are no peers yet, there are only local transactions, correct? Therefore, instead of having different eviction policies, could we make time-based eviction only occur after the `MAX_PENDING_TRANSACTIONS` has been reached?

Reply:
It seems that i misunderstood Adrian's suggestion. It was more about processing incoming messages from the network that add transactions to the pool. We can still use the eviction mechanism to fix this but we should probably also limit the number of messages we handle asynchronously.